org.eclipse.ice.dev/org.eclipse.ice.dev.pojofromjson/src/main/java/org/eclipse/ice/dev/pojofromjson/PojoFromJson.java

```diff
@@ -76,38 +76,47 @@ public static void handleInputJson(
 	) throws JsonParseException, JsonMappingException, IOException {
 		// Parse outline from input stream
 		PojoOutline pojo = mapper.readValue(is, PojoOutline.class);
-
+		createDataElement(pojo, destination);
+	}
+	
+	/**
+	 * Writes the java interface and implementation files of given pojo outline in destination
+	 * @param pojo PojoOutline representing the JSON
+	 * @param destination directory in which files will be generated
+	 * @throws IOException On failure to open file for writing 
+	 */
+	public static void createDataElement(PojoOutline pojo, Path destination) throws IOException {
 		// Collect fields
 		Fields fields = new Fields();
 		fields.collect(DefaultFields.get());
 		fields.collect(pojo.getFields());
-
+		
 		// Write Interface
 		try (Writer elementInterface = Files.newBufferedWriter(
-			destination.resolve(pojo.getElement() + ".java")
-		)) {
+				destination.resolve(pojo.getElement() + ".java")
+				)) {
 			InterfaceWriter.builder()
-				.packageName(pojo.getPackageName())
-				.interfaceName(pojo.getElement())
-				.fields(fields)
-				.types(new Types(fields.getInterfaceFields()))
-				.build()
-				.write(elementInterface);
+			.packageName(pojo.getPackageName())
+			.interfaceName(pojo.getElement())
+			.fields(fields)
+			.types(new Types(fields.getInterfaceFields()))
+			.build()
+			.write(elementInterface);
 		}
 
 		// Write implementation
 		try (Writer elementImpl = Files.newBufferedWriter(
-			destination.resolve(pojo.getImplementation() + ".java")
-		)) {
-			ImplementationWriter.builder()
-				.packageName(pojo.getPackageName())
-				.interfaceName(pojo.getElement())
-				.className(pojo.getImplementation())
-				.fields(fields)
-				.types(fields.getTypes())
-				.build()
-				.write(elementImpl);
-		}
+				destination.resolve(pojo.getImplementation() + ".java")
+					)) {
+						ImplementationWriter.builder()
+							.packageName(pojo.getPackageName())
+							.interfaceName(pojo.getElement())
+							.className(pojo.getImplementation())
+							.fields(fields)
+							.types(fields.getTypes())
+							.build()
+							.write(elementImpl);
+					}
```

Comment:
Seems to be some strange formatting stuff going on here.

Reply:
Yeah, sometimes weird things happen with formatting when I'm moving pieces of code.